labelbox/schema/dataset.py

```diff
@@ -58,12 +60,75 @@ class Dataset(DbObject, Updateable, Deletable):
         deprecation_warning=
         "This method does not return any data for batch-based projects and it will be deprecated on or around November 1, 2023."
     )
-    data_rows = Relationship.ToMany("DataRow", False)
     created_by = Relationship.ToOne("User", False, "created_by")
     organization = Relationship.ToOne("Organization", False)
     iam_integration = Relationship.ToOne("IAMIntegration", False,
                                          "iam_integration", "signer")
 
+    def data_rows(
+        self,
+        from_cursor: str = None,
+        where: Comparison = None,
+    ) -> PaginatedCollection:
+        """ 
+        Custom method to paginate data_rows via cursor.
+
+        Params:
+            after (str): Cursor (data row id) to start from, if none, will start from the beginning
+            where (dict(str,str)): Filter to apply to data rows. Where value is a data row column name and key is the value to filter on.    
+                example: {'external_id': 'my_external_id'} to get a data row with external_id = 'my_external_id'
+
+
+        NOTE: 
+            Order of retrieval is newest data row first.
+            Deleted data rows are not retrieved.
+            Failed data rows are not retrieved.
+            Data rows in progress *maybe* retrieved.
+        """
+
+        page_size = 500  # hardcode to avoid overloading the server
```

Comment:
should we rather define this in the backend? the users can overwrite this anytime to cause overload

Reply:
I see that... the problem was that currently 100 is hardcoded by our paginator for all other cases... let me look into it

Reply:
but I have looked at it again and the thing is, we have a const `_PAGE_SIZE` explicitly setting page size in `PaginatedCollection`. So, I guess I could add some param to '`not_set_page_size` and use it for this particular case... I know though we have a separate story to review pagination limits so perhaps I do it there?

Reply:
Since PAGE_SIZE is already in the SDK, let's not remove it for now, since it'd be a breaking change